tasks.c

```diff
@@ -6615,8 +6615,16 @@ static void prvResetNextTaskUnblockTime( void )
                 pcWriteBuffer = prvWriteNameToBuffer( pcWriteBuffer, pxTaskStatusArray[ x ].pcTaskName );
 
                 /* Write the rest of the string. */
-                sprintf( pcWriteBuffer, "\t%c\t%u\t%u\t%u\r\n", cStatus, ( unsigned int ) pxTaskStatusArray[ x ].uxCurrentPriority, ( unsigned int ) pxTaskStatusArray[ x ].usStackHighWaterMark, ( unsigned int ) pxTaskStatusArray[ x ].xTaskNumber ); /*lint !e586 sprintf() allowed as this is compiled with many compilers and this is a utility function only - not part of the core kernel implementation. */
-                pcWriteBuffer += strlen( pcWriteBuffer );                                                                                                                                                                                                /*lint !e9016 Pointer arithmetic ok on char pointers especially as in this case where it best denotes the intent of the code. */
+                #ifndef configTASK_WRITE_BUFFER_LENGTH
```

Comment:
To reduce conditional code, `configTASK_WRITE_BUFFER_LENGTH` could have a default value of SIZE_MAX if the developer doesn't define it.  Actually, it seems SIZE_MAX was introduced in C99, so that symbol would need to be defined conditionally first, e.g., `( (size_t) -1 )`.

As a (better?) alternative, if `configUSE_STATS_FORMATTING_FUNCTIONS > 0` then we could generate a compile-time error if `configTASK_WRITE_BUFFER_LENGTH` is not defined.  This is not backward compatible but perhaps is justified.  Would need direction from the FreeRTOS team on this question.

Reply:
The current thought is that we don't want to break backwards compatibility (demo code maybe using this function and we want to minimize the impact to that), so I think adding just a config for this is fine.